eth/downloader/downloader.go

```diff
@@ -470,12 +470,10 @@ func (d *Downloader) cancel() {
 		}
 	}
 	d.cancelLock.Unlock()
-
-	// Reset the queue
-	d.queue.Reset()
 }
 
 // Terminate interrupts the downloader, canceling all pending operations.
+// The downloader cannot be used after calling Terminate.
```

Comment:
Nitpicking, but I would change `used` to `reused` to emphasize that you have to throw it away completely.